tests/phpunit/tests/blocks/wpGetBlockPatterns.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Tests for _wp_get_block_patterns.
+ *
+ * @package WordPress
+ * @subpackage Blocks
+ * @since 6.4.0
+ *
+ * @group blocks
+ *
+ * @covers ::_wp_get_block_patterns
+ */
+class Tests_Blocks_WpGetBlockPatterns extends WP_UnitTestCase {
+	/**
+	 * @ticket 59490
+	 *
+	 * @dataProvider data_wp_get_block_patterns
+	 *
+	 * @param string $theme    The theme's slug.
+	 * @param array  $expected The expected pattern data.
+
+	 */
+	public function test_should_return_block_patterns( $theme, $expected ) {
+		$patterns = _wp_get_block_patterns( wp_get_theme( $theme ) );
+		$this->assertSameSets( $expected, $patterns );
+	}
+
+	/**
+	 * @ticket 59490
+	 */
+	public function test_delete_theme_cache() {
+		$theme = wp_get_theme( 'block-theme-patterns' );
+		_wp_get_block_patterns( $theme );
+		$transient = get_site_option( 'wp_theme_patterns_block-theme-patterns' );
```

Comment:
Here and throughout the tests, they would need to be updated to use transients accordingly as well.